packages/typespec-python/test/azure/generated/azure-mgmt-spheredpg/azure/mgmt/spheredpg/_model_base.py

```diff
@@ -560,6 +572,27 @@ def _as_dict_value(v: typing.Any, exclude_readonly: bool = False) -> typing.Any:
             return {dk: Model._as_dict_value(dv, exclude_readonly=exclude_readonly) for dk, dv in v.items()}
         return v.as_dict(exclude_readonly=exclude_readonly) if hasattr(v, "as_dict") else v
 
+    def __getattr__(self, name: str) -> typing.Any:
+        if not name.startswith("__"):
+            if "properties" in self._attr_to_rest_field:
+                if self.properties:
+                    return getattr(self.properties, name)
+                class_type = self._attr_to_rest_field["properties"]._class_type
+                if name in getattr(class_type, "__dict__", {}):
+                    return None
+        raise AttributeError(f"'{self.__class__.__name__}' object has no attribute '{name}'")
+
+    def __setattr__(self, key: str, value: typing.Any) -> None:
+        if key != "properties" and "properties" in self._attr_to_rest_field:
+            class_type = self._attr_to_rest_field["properties"]._class_type
+            properties_attr_dict = getattr(class_type, "__dict__", {})
```

Comment:
nit: I think it might be clearer if we do something like this, at least for me. I find the `getattr(..., "__dict__", {}) a little confusing. 

```python
try:
  getattr(self.properties, key)
  if getattr(self, "properties", None) is None:
    setattr(self, "properties", class_type())
  setattr(self.properties, key, value)
  return
except AttributeError:
  pass
super().__setattr__(key, value)
```